test/redis-integration-test.ts

```diff
@@ -0,0 +1,251 @@
+import process from 'node:process'
+import { describe, it, expect, beforeAll, afterAll, jest } from '@jest/globals'
+import { Redis, Cluster } from 'ioredis'
+import { type Options as RateLimitOptions } from 'express-rate-limit'
+import { RedisStore } from '../source/index.js'
+import type {
+	SendCommandClusterDetails,
+	RedisReply,
+	Options as RedisOptions,
+} from '../source/types.js'
+
+jest.setTimeout(30_000)
+
+describe('Redis Integration Tests', () => {
+	describe('Single Redis Instance', () => {
+		let client: Redis
+		let store: RedisStore
+
+		beforeAll(async () => {
+			const host = process.env.REDIS_HOST ?? 'localhost'
+			const port = Number(process.env.REDIS_PORT ?? 6385)
+
+			client = new Redis({
+				host,
+				port,
+				lazyConnect: true,
+			})
+			await client.connect().catch(() => {
+				console.warn('Skipping Redis tests - connection failed')
+			})
```

Comment:
I think I'd rather have the tests fail if it can't connect

```suggestion
			await client.connect()
```

I *think* jest will skip the rest of the tests automatically if `beforeAll` fails.

Reply:
Fix: https://github.com/express-rate-limit/rate-limit-redis/pull/225/commits/70e6107aecf3482579b86d961a1f049f242f0e66